op-e2e/actions/l2_engine_api.go

```diff
@@ -42,13 +42,13 @@ func computePayloadId(headBlockHash common.Hash, params *eth.PayloadAttributes)
 	_ = binary.Write(hasher, binary.BigEndian, params.Timestamp)
 	hasher.Write(params.PrevRandao[:])
 	hasher.Write(params.SuggestedFeeRecipient[:])
+	_ = binary.Write(hasher, binary.BigEndian, params.NoTxPool)
+	_ = binary.Write(hasher, binary.BigEndian, uint64(len(params.Transactions)))
 	for _, tx := range params.Transactions {
 		_ = binary.Write(hasher, binary.BigEndian, uint64(len(tx))) // length-prefix to avoid collisions
 		hasher.Write(tx)
 	}
-	if params.NoTxPool {
-		hasher.Write([]byte{1})
-	}
+	_ = binary.Write(hasher, binary.BigEndian, *params.GasLimit)
```

Comment:
Do you need to do a nil check here?

Reply:
it's always L2 here, and just quick testing code while we wait for the random ID. In op geth I have the nil check.